Lib/test/test_importlib/import_/test___loader__.py

```diff
@@ -23,6 +23,10 @@ def test___loader__(self):
         with util.uncache('blah'), util.import_state(meta_path=[loader]):
             module = self.__import__('blah')
         self.assertEqual(loader, module.__loader__)
+        expected_repr_pattern = (
+            r"<module 'blah' \(<test\.test_importlib\..*SpecLoaderMock object at .+>\)>"
+        )
+        self.assertRegex(repr(module), expected_repr_pattern)
```

Comment:
FWIW, I'd expect this in a distinct test with the other repr-related tests in `Lib/test/test_module/__init__.py`.

Reply:
Thanks! I'm new to hacking on `importlib`; didn't realise there was a more appropriate place for this test to go :)

I'm happy to move this into `Lib/test/test_module/__init__.py` in a followup PR, but I might need some guidance on how to construct a test like this without the help of the `test_importlib.util` functions used in this test case. (I assume it would be bad to import `test_importlib.util` inside `test_module/__init__.py`...?)

Reply:
> Thanks! I'm new to hacking on importlib; I didn't realise there was a more appropriate place for this test to go :)

Don't sweat it.  Thanks for being willing to dive into something unfamiliar!  Not knowing things is especially typical when you do that. 😄

> I'm happy to move this into Lib/test/test_module/__init__.py in a followup PR, but I might need some guidance on how to construct a test like this

The pattern of the other repr-related tests there is to create a dummy module object with the various pieces set manually.  In this case there would be two tests:

```py
    def test_module_repr_with_namespace_package(self):
        m = ModuleType('foo')
        m.__loader__ = importlib.machinery.NamespaceLoader(...)
        ...
        self.assertEqual(repr(m), '...')

    def test_module_repr_with_fake_namespace_package(self):
        m = ModuleType('foo')
        m.__loader__ = BareLoader()
        m.__loader__._path = ['spam']
        self.assertEqual(repr(m), '...')
```

Reply:
FWIW, it's also nice to have functional tests that don't bake implementation details into the tests so much, though in this case it isn't so bad.

We do have functional repr-related tests like that for at least some of the specific module types (e.g. frozen, builtin).  There is even one already for namespace packages in Lib/test/test_importlib/test_namespace_pkgs.py.  It might be nice to have one in that file that tests the looks-like-a-namespace-package case functionally.  However, I don't think that's critical.

Reply:
Fab. I filed #112475 to add the tests in the right place. Thanks for the help!